File: Python/Semi-Colon E-commerce/CartItem.py
class CartItem:
    def __init__(self, product, product_count):
        self._product = product
        self._product_count = product_count

    def get_product_name(self):
        return self._product

    def get_product_count(self):
        return self._product_count

    def add_product_count(self, product_quantity):
        self._product_count += product_quantity

    def get_item_total(self):
        return self._product.get_price() * self._product_count

    def get_item_discount(self, discount_rate):
        return self._product.get_price() * discount_rate

File: Java/Semi-Colon E-commerce/src/CartItem.java
public class CartItem {
    private final Product product;
    private int productCount;

    public CartItem(Product product, int productCount) {
        this.product = product;
        this.productCount = productCount;
    }

    public Product getProductName() {
        return product;
    }

    public int getProductCount() {
        return productCount;
    }

    public void addProductCount(int productQuantity) {
        productCount += productQuantity;
    }

    public double getItemTotal() {
        return product.getPrice() * productCount;
    }

    public double getItemDiscount(double discountRate) {
        return product.getPrice() * discountRate;
    }
}
